lib/processing.py

```diff
@@ -0,0 +1,490 @@
+import lib.utils as utils
+import re
+import json
+import cv2
+import fitz 
+import tiktoken
+import pytesseract
+import numpy as np
+from PIL import Image
+from openai import OpenAI
+from qdrant_client import QdrantClient
+from pdf2image import convert_from_path
```

Comment:
I know this is the first library you find when looking for a way to turn a PDF into images but is way too slow and consumes a lot of resources. I ended up doing my own implementation using a PDF management library